source/Calamari.Aws/DeleteCloudFormationCommand.cs

```diff
@@ -1,57 +1,35 @@
 ﻿using System;
-using System.Collections.Generic;
-using System.IO;
-using Amazon.CloudFormation;
+using Amazon.IdentityManagement;
+using Amazon.SecurityToken;
 using Calamari.Aws.Deployment;
-using Calamari.Aws.Deployment.Conventions;
+using Calamari.Aws.Deployment.CloudFormation;
 using Calamari.Aws.Integration.CloudFormation;
-using Calamari.Aws.Util;
-using Calamari.CloudAccounts;
 using Calamari.Commands.Support;
 using Calamari.Deployment;
-using Calamari.Deployment.Conventions;
-
-namespace Calamari.Aws.Commands
+namespace Calamari.Aws
 {
     [Command(AwsKnownVariables.Commands.DeleteAwsCloudformation, Description = "Destroy an existing AWS CloudFormation stack")]
-    public class DeleteCloudFormationCommand : Command
+    public class DeleteCloudFormationCommand : AwsCommand
     {
-        readonly ILog log;
-        readonly IVariables variables;
-        private string packageFile;
-        private bool waitForComplete;
-        
-        public DeleteCloudFormationCommand(ILog log, IVariables variables)
+        readonly ICloudFormationService cloudFormationService;
+
+        public DeleteCloudFormationCommand(
+            ILog log,
+            IVariables variables,
+            IAmazonSecurityTokenService amazonSecurityTokenService,
+            IAmazonIdentityManagementService amazonIdentityManagementService,
+            ICloudFormationService cloudFormationService)
+            : base(log, variables, amazonSecurityTokenService, amazonIdentityManagementService)
         {
-            this.log = log;
-            this.variables = variables;
-            Options.Add("package=", "Path to the NuGet package to install.", v => packageFile = Path.GetFullPath(v));
-            Options.Add("waitForCompletion=", "True if the deployment process should wait for the stack to complete, and False otherwise.", v => waitForComplete =  
-                !bool.FalseString.Equals(v, StringComparison.OrdinalIgnoreCase)); //True by default
+            this.cloudFormationService = cloudFormationService;
         }
 
-        public override int Execute(string[] commandLineArguments)
+        protected override void Execute(RunningDeployment deployment)
         {
-            Options.Parse(commandLineArguments);
+            var stackArn = new StackArn(deployment.Variables.Get(AwsSpecialVariables.CloudFormation.StackName));
+            var waitForCompletion = !bool.FalseString.Equals(variables.Get("waitForCompletion"), StringComparison.OrdinalIgnoreCase);
```

Comment:
The `VariableDictionary` in `OctoStache` has a function for this `EvaluateTruthy` that you could use here

Reply:
Noted will change in a Later PR